### PR TITLE
docs: create migration guide for v4

### LINKED ISSUE #2935: Documentation for users migrating to Webpack v5
- Operating System: N/A
- Node Version: N/A
- NPM Version: N/A
- webpack Version: 5.10.1
- webpack-dev-server Version: 3.11.0
- Browser: N/A

- [ ] This is a **bug**
- [ ] This is a **modification** request
- [X] This is a **documentation** request

I just upgraded from Webpack v4 to Webpack v5.  From other issues on the tracker, it looks like I'm supposed to upgrade to `webpack-dev-server@4-beta`, but there is no mention of the Webpack 4-to-5 transition, either on the front page of this repo, or on the [dev-server docs at js.org](https://webpack.js.org/configuration/dev-server/#devserver), or in the [v5 migration guide](https://webpack.js.org/migrate/5/).

Webpack v5 is a big breaking change for most users.  If the 3.x branch of `webpack-dev-server` isn't going to be supported under Webpack v5, the docs should clearly say that, in at least one of the places mentioned above.  (If 3.x continues to be officially supported under v5, it wouldn't hurt to say that explicitly as well.)

### PR DESCRIPTION
**Pull request by snitin315**
- [ ] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [X] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?
NA

### Motivation / Use-Case
add migration guide for users.

### Breaking Changes
No

### Additional Info

fixes #2935

### CODE REVIEW

**Review by alexander-akait — COMMENTED** [review]
Good start, let's describe:
- `allowedHosts` and `disableHostCheck` (was removed in favor `allowedHosts: 'all'`). 
- `useLocalIp` was removed in favor `host: 'local-ip'` (`local-ipv4` and `local-ipv6`)
- `warn` was removed in favor https://webpack.js.org/configuration/other-options/#ignorewarnings
- `contentBase`/`contentBasePublicPath`/`watchContentBase` in favor `static`
- `transportMode` was removed in favor `webSocketServer` and it is `ws` by default
- `quiet`/`noInfo` was removed in favor built-in logger and can be reconfigured using https://webpack.js.org/configuration/other-options/#infrastructurelogging

**Comment by snitin315** [comment]
Take a look at https://github.com/webpack/webpack-dev-server/blob/migration-guide/migration-v4.md

File: migration-v4.md [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+        res.json({ custom: 'response' });
+      });
+    },
+  },
+};
+```
+
+## `client` options
+
+The `sockHost`, `sockPath`, `sockPort`, `clientLogLevel`, `injectClient`, `injectHot`, `overlay`, and `progress` options were moved to `client` option:
```
Comment (alexander-akait) [inline migration-v4.md:47]: Let's describe here, `sockHost`, `sockPath` and `sockPort` was merged into `webSocketURL`

**Review by alexander-akait — COMMENTED** [review]
Also let's describe:
- `filename`, `lazy` was removed in favor https://webpack.js.org/configuration/experiments/#experimentslazycompilation
- `hot` is `true` by default
- `compress` is `true` by default
- `inline` was removed without replacement
- `log`, `logLevel` and `logTime`  was removed without replacement, now we use built-in logger
- `overlay` is true by default
- `public` was removed in favor `client.webSocketURL`

Ideally we should describe all removed/changed options from here https://github.com/webpack/webpack-dev-server/blob/v3.11.2/lib/options.json

**Comment by alexander-akait** [comment]
@snitin315 I will merge it after CI green, if you found typo or if I missed something feel free to send a PR

**Pull request merged**